Linux build fails with musl libc because struct mallinfo is not available

Linux build fails with musl libc

At

https://github.com/project-chip/connectedhomeip/blob/bc4d7114c60d40d517bdd9da7d2f51c00e2a70ee/src/platform/Linux/PlatformManagerImpl.cpp#L154

with

```log
../../third_party/connectedhomeip/src/platform/Linux/PlatformManagerImpl.cpp:154:34: error: invalid use of incomplete type 'mallinfo'
    struct mallinfo mallocInfo = mallinfo();
                                 ^~~~~~~~~~
../../third_party/connectedhomeip/src/platform/Linux/PlatformManagerImpl.cpp:154:12: note: forward declaration of 'mallinfo'
    struct mallinfo mallocInfo = mallinfo();
           ^
```

Possible solution: Add build arg to bypass usage of mallinfo.

--- a/src/platform/Linux/DiagnosticDataProviderImpl.cpp
+++ b/src/platform/Linux/DiagnosticDataProviderImpl.cpp
@@ -218,6 +218,9 @@ DiagnosticDataProviderImpl & DiagnosticDataProviderImpl::GetDefaultInstance()
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapFree(uint64_t & currentHeapFree)
 {
+#ifndef __GLIBC__
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+#else
     struct mallinfo mallocInfo = mallinfo();
 
     // Get the current amount of heap memory, in bytes, that are not being utilized
@@ -225,10 +228,14 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapFree(uint64_t & currentHeap
     currentHeapFree = mallocInfo.fordblks;
 
     return CHIP_NO_ERROR;
+#endif
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapUsed(uint64_t & currentHeapUsed)
 {
+#ifndef __GLIBC__
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+#else
     struct mallinfo mallocInfo = mallinfo();
 
     // Get the current amount of heap memory, in bytes, that are being used by
@@ -236,10 +243,14 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapUsed(uint64_t & currentHeap
     currentHeapUsed = mallocInfo.uordblks;
 
     return CHIP_NO_ERROR;
+#endif
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & currentHeapHighWatermark)
 {
+#ifndef __GLIBC__
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+#else
     struct mallinfo mallocInfo = mallinfo();
 
     // The usecase of this function is embedded devices,on which we would need to intercept
@@ -252,6 +263,7 @@ CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapHighWatermark(uint64_t & cu
     currentHeapHighWatermark = mallocInfo.uordblks;
 
     return CHIP_NO_ERROR;
+#endif
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetThreadMetrics(ThreadMetrics ** threadMetricsOut)

